app/controllers/api/landing_applications_controller.rb

```diff
@@ -1,7 +1,19 @@
 class Api::LandingApplicationsController < ApplicationController
+  before_action :authenticate_api_key
+
   def index
     render json: LandingApplication.includes(:destination).map { |la|
       LandingApplicationEntity.new(la).represent
     }
   end
+
+  private
+
+  def authenticate_api_key
+    head :unauthorized unless ApiClientAuthenticator.authenticate?(api_key)
+  end
+
+  def api_key
+    request.headers["HTTP_X_API_KEY"]
+  end
```

Comment:
how much value do you think there is in having this as a separate method? Is it not quite clear already from the name?

Reply:
fair point, I guess it's a marginal win in terms of reducing cognitive load